### PR TITLE
BaseAbstractTemplate: add support for locale

### PR DESCRIPTION
**Pull request by xkubov**

### CODE REVIEW

File: pymess/migrations/0030_migration.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+            old_name='template',
+            new_name='template_old',
+        ),
+        migrations.RenameField(
+            model_name='smstemplatedisallowedobject',
+            old_name='template',
+            new_name='template_old',
+        ),
+
+        # We add indexes to speed-up insertion.
```
Comment (xkubov) [inline pymess/migrations/0030_migration.py]: The migration hit our one-hour limit during testing. I added indexes to see if they would help, but they didn't.

I'm open to suggestions on how to make this migration more efficient.

File: pymess/migrations/0030_migration.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
+# Generated by Django 4.2.13 on 2024-07-03 18:01
+
+import django.db.models.deletion
+from django.db import migrations, models
+from utils.migrations import migration_print
```
Comment (xkubov) [inline pymess/migrations/0030_migration.py]: I'll remove these prints before merging.

File: pymess/management/commands/sync_emails.py [hunk 3]
```diff
@@ -34,3 +31,15 @@ def handle(self, *args, **options):
...
                 self.style.SUCCESS('Synced "{}" e-mail templates'.format(len(files_with_email_html_body)))
             )
+
+    def handle(self, *args, **options):
+        verbosity = int(options.get('verbosity'))
+        if verbosity > 0:
+            self.stdout.write('Syncing e-mails')
+
+        directory = os.path.join(settings.EMAIL_HTML_DATA_DIRECTORY)
+        for filename in os.listdir(directory):
```
Comment (Formulka) [inline pymess/management/commands/sync_emails.py:43]: `filename` is actually a directory name (or should be), specifically the locale directory name. Maybe call it `locale_directory`?

File: pymess/models/emails.py [hunk 4]
```diff
@@ -277,6 +277,7 @@ class Meta(BaseRelatedObject.Meta):
 class EmailTemplateAttachment(SmartModel):
 
     template = models.ForeignKey(to=settings.EMAIL_TEMPLATE_MODEL, verbose_name=_('template'),
+                                 null=True, blank=True,
```
Comment (Formulka) [inline pymess/models/emails.py]: this should probably stay null/blank False

**Review by Formulka — APPROVED** [review]

File: pymess/models/emails.py (hunk 4 above)
Comment (xkubov) [reply pymess/models/emails.py]: I added this to be consistent with all `Message` classes (e.g., `EmailMessage`). On the other hand, I get that for this model it doesn't make much sense. I was conflicted on whether to make the extra effort as the null/blank=True are needed for the migration but I can make them `False` again at the end of the migration.

What do you think @rubickcz

File: pymess/management/commands/sync_emails.py (hunk 3 above)
Comment (xkubov) [reply pymess/management/commands/sync_emails.py:43]: I've added comments to clarify my naming. The root dir may contain `.html` files that would be synced for a locale with `None` value.

File: pymess/migrations/0030_migration.py (hunk 2 above)
Comment (rubickcz) [reply pymess/migrations/0030_migration.py]: I think you can do it.

File: pymess/models/emails.py (hunk 4 above)
Comment (rubickcz) [reply pymess/models/emails.py]: This object makes no sense without reference to a template. So I agree with @Formulka.

**Review by rubickcz — APPROVED** [review]

**Pull request merged**